src/main/java/core/basesyntax/data/servise/FruitShopService.java

```diff
@@ -0,0 +1,42 @@
+package core.basesyntax.data.servise;
+
+import core.basesyntax.data.exeption.UnknownOperationException;
+import core.basesyntax.data.model.FruitTransaction;
+import core.basesyntax.data.model.OperationType;
+import core.basesyntax.data.strategy.BalanceOperationHandler;
+import core.basesyntax.data.strategy.OperationHandler;
+import core.basesyntax.data.strategy.PurchaseOperationHandler;
+import core.basesyntax.data.strategy.ReturnOperationHandler;
+import core.basesyntax.data.strategy.SupplyOperationHandler;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class FruitShopService {
+    private Map<String, Integer> fruitInventory = new HashMap<>();
+    private Map<OperationType, OperationHandler> operationStrategy = new HashMap<>();
+
+    public FruitShopService(Map<OperationType, OperationHandler> operationStrategy) {
+        this.operationStrategy.put(OperationType.BALANCE, new BalanceOperationHandler());
+        this.operationStrategy.put(OperationType.SUPPLY, new SupplyOperationHandler());
+        this.operationStrategy.put(OperationType.PURCHASE, new PurchaseOperationHandler());
+        this.operationStrategy.put(OperationType.RETURN, new ReturnOperationHandler());
+    }
```

Comment:
The constructor initializes the `operationStrategy` map with new handlers, even though the map is passed as a parameter. This makes the parameter redundant. Consider using the provided `operationStrategy` map directly or removing the parameter if it's not intended to be used.